src/biotite/structure/io/pdbx/cif.py

```diff
@@ -1024,6 +1029,34 @@ def _multiline(value):
     return value
 
 
+def _split_one_line(line):
+    """
+    Split a line into its fields.
+    Supporting embedded quotes (' or "), like `'a dog's life'` to  `a dog's life`
+    """
+    # Define the patterns for different types of fields
+    single_quote_pattern = r"('(?:'(?! )|[^'])*')(?:\s|$)"
+    double_quote_pattern = r'("(?:"(?! )|[^"])*")(?:\s|$)'
+    unquoted_pattern = r"([^\s]+)"
```

Comment:
Thanks for the explanation of the regex patterns in the PR :+1:. I tested the patterns on https://regex101.com/ and they seem to capture all fields correctly in my test cases.

However, the current pattern captures the quotes as well, requiring the potentially time consuming quote removal below. Hence, I suggest a modified pattern, that makes the 'quoted group' non-capturing and captures a new group excluding the quotes instead. This way the quote removal is directly part of the regex pattern.

```suggestion
    single_quote_pattern = r"(?:'((?:'(?! )|[^'])*)')(?:\s|$)"
    double_quote_pattern = r'(?:"((?:"(?! )|[^"])*)")(?:\s|$)'
    unquoted_pattern = r"([^\s]+)"
```

Reply:
Capturing quotes for handling emtpy string (`''`, `""`), I try not capture quotes and add two regex pattern for empty string, it is slower, see `re_split_empty(line)`
Quote removal cost about 2/13 time of `_split_one_line()`, see `re_split_no_quote_removal(line)`.

I did not figure out faster way to handle empty string, so I leave the regex pattern unchanged.

- Test result:
```python
                line='\'\' ATOM      1  N   PRO A   1   \'\'  ""  \'1.400\'  "10.00"  10.000  1.00  0.00           N  ""'
   shlex.split(line)=['', 'ATOM', '1', 'N', 'PRO', 'A', '1', '', '', '1.400', '10.00', '10.000', '1.00', '0.00', 'N', '']
      re_split(line)=['', 'ATOM', '1', 'N', 'PRO', 'A', '1', '', '', '1.400', '10.00', '10.000', '1.00', '0.00', 'N', '']
re_split_empty(line)=['', 'ATOM', '1', 'N', 'PRO', 'A', '1', '', '', '1.400', '10.00', '10.000', '1.00', '0.00', 'N', '']
run number=1000000 times timeit.timeit(                'str_split(line)', number=number, globals=locals()) = 1.9387128870002925 s
run number=1000000 times timeit.timeit(              'shlex.split(line)', number=number, globals=locals()) = 29.21148640103638 s
run number=1000000 times timeit.timeit(                 're_split(line)', number=number, globals=locals()) = 13.35447793500498 s
run number=1000000 times timeit.timeit(           're_split_empty(line)', number=number, globals=locals()) = 14.940637337975204 s
run number=1000000 times timeit.timeit(         're_complie_split(line)', number=number, globals=locals()) = 13.758984111249447 s
run number=1000000 times timeit.timeit('re_split_no_quote_removal(line)', number=number, globals=locals()) = 11.339349532034248 s
```

<details>
  <summary>benchmark code:</summary>

import shlex
import re

def str_split(line):
    values = line.split()
    for k in range(len(values)):
        # Remove quotes
        if (values[k][0] == '"' and values[k][-1] == '"') or (
            values[k][0] == "'" and values[k][-1] == "'"
    ):
            values[k] = values[k][1:-1]


def re_split(line):
    """
    Split a line into its fields.
    Supporting embedded quotes (' or "), like `'a dog's life'` to  `a dog's life`
    """
    # Define the patterns for different types of fields
    single_quote_pattern = r"('(?:'(?! )|[^'])*')(?:\s|$)"
    double_quote_pattern = r'("(?:"(?! )|[^"])*")(?:\s|$)'
    unquoted_pattern = r"([^\s]+)"

    # Combine the patterns using alternation
    combined_pattern = (
        f"{single_quote_pattern}|{double_quote_pattern}|{unquoted_pattern}"
    )

    # Find all matches
    matches = re.findall(combined_pattern, line)

    # Extract non-empty groups from the matches
    fields = []
    for match in matches:
        field = next(group for group in match if group)
        if field[0] == field[-1] == "'" or field[0] == field[-1] == '"':
            field = field[1:-1]
        fields.append(field)
    return fields

def re_split_no_quote_removal(line):
    """
    Split a line into its fields.
    Supporting embedded quotes (' or "), like `'a dog's life'` to  `a dog's life`
    """
    # Define the patterns for different types of fields
    single_quote_pattern = r"('(?:'(?! )|[^'])*')(?:\s|$)"
    double_quote_pattern = r'("(?:"(?! )|[^"])*")(?:\s|$)'
    unquoted_pattern = r"([^\s]+)"

    # Combine the patterns using alternation
    combined_pattern = (
        f"{single_quote_pattern}|{double_quote_pattern}|{unquoted_pattern}"
    )

    # Find all matches
    matches = re.findall(combined_pattern, line)

    # Extract non-empty groups from the matches
    fields = []
    for match in matches:
        field = next(group for group in match if group)
        fields.append(field)
    return fields

# Define the patterns for different types of fields
_single_quote_pattern = r"('(?:'(?! )|[^'])*')(?:\s|$)"
_double_quote_pattern = r'("(?:"(?! )|[^"])*")(?:\s|$)'
_unquoted_pattern = r"([^\s]+)"

_SPLIT_FIELD_PATTERN = re.compile(
        f"{_single_quote_pattern}|{_double_quote_pattern}|{_unquoted_pattern}"
    )

def re_complie_split(line):
    """
    Split a line into its fields.
    Supporting embedded quotes (' or "), like `'a dog's life'` to  `a dog's life`
    """

    # Find all matches
    matches = re.findall(_SPLIT_FIELD_PATTERN, line)

    # Extract non-empty groups from the matches
    fields = []
    for match in matches:
        field = next(group for group in match if group)
        if field[0] == field[-1] == "'" or field[0] == field[-1] == '"':
            field = field[1:-1]
        fields.append(field)
    return fields


def re_split_empty(line):
    """
    Split a line into its fields.
    Supporting embedded quotes (' or "), like `'a dog's life'` to  `a dog's life`
    """
    # Define the patterns for different types of fields
    single_empty_pattern = r"(?:\s|^)('')(?:\s|$)"
    double_empty_pattern = r'(?:\s|^)("")(?:\s|$)'
    single_quote_pattern = r"'((?:'(?! )|[^'])*)'(?:\s|$)"
    double_quote_pattern = r'"((?:"(?! )|[^"])*)"(?:\s|$)'
    unquoted_pattern = r"([^\s]+)"

    # Combine the patterns using alternation
    combined_pattern = (
        f"{single_empty_pattern}|{double_empty_pattern}|{single_quote_pattern}|{double_quote_pattern}|{unquoted_pattern}"
    )

    # Find all matches
    matches = re.findall(combined_pattern, line)

    # Extract non-empty groups from the matches
    fields = []
    for match in matches:
        # print(list(match))
        field = next(group for group in match if group)
        if field == "''" or field == '""':
            field = ""
        fields.append(field)
    return fields


if __name__ == '__main__':
    import timeit

    line = """'' ATOM      1  N   PRO A   1   ''  ""  '1.400'  "10.00"  10.000  1.00  0.00           N  """ + '""'

    print(f"{                line=}")
    print(f"{   shlex.split(line)=}")
    print(f"{      re_split(line)=}")
    print(f"{re_split_empty(line)=}")
    number = 1000000
    # number = 1000
    print(f"run {number=} times {timeit.timeit(                'str_split(line)', number=number, globals=locals()) = } s")
    print(f"run {number=} times {timeit.timeit(              'shlex.split(line)', number=number, globals=locals()) = } s")
    print(f"run {number=} times {timeit.timeit(                 're_split(line)', number=number, globals=locals()) = } s")
    print(f"run {number=} times {timeit.timeit(           're_split_empty(line)', number=number, globals=locals()) = } s")
    print(f"run {number=} times {timeit.timeit(         're_complie_split(line)', number=number, globals=locals()) = } s")
    print(f"run {number=} times {timeit.timeit('re_split_no_quote_removal(line)', number=number, globals=locals()) = } s")

</details>

Reply:
I also tried my luck a bit on increasing the performance, but still your implementation is the fastest :+1:.